### PR TITLE
Polio-675, polio-668: numbers separator

### PR DESCRIPTION
**Pull request by quang-le**
Format the numbers to have thousands and decimal separator in vaccine tab and budget workflow

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [X] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests
## Changes

- Budget: 
    - used `NumberInput` for the amount field
    - used `convertToNumber` to convert number with separator before sending to API
    - adapted yup validation: created new custom method
- Vaccine tab:
    -  Added `NumberInput` in polio that can be used as child of formik's `Field`
    -  Used it for shipments, form a fields and destructions
    -  Adapted `UseSaveCampaign` to convert the values back to numbers before sending to the backend
    -  Adapted yup validation 

## How to test

Budget: go to any budget (with the right setup and permissions) and play with the "amount" field
Vaccine tab: likewise: test the fields that can be filled with numbers for form a, destruction and shipment in vaccine tab (ie all fields that are not date, select or comment)

## Print screen / video

https://user-images.githubusercontent.com/38907762/204022987-c0d9bbc5-221b-416b-b6a1-2c37c6ac55f6.mov

## Notes

 ⚠️ Needs https://github.com/BLSQ/bluesquare-components/pull/90

### CODE REVIEW

File: plugins/polio/js/src/hooks/useSaveCampaign.js [hunk 1]
```diff
@@ -8,6 +10,71 @@ const formatGroupedCampaigns = groupedCampaigns => {
         return commaSeparatedIdsToStringArray(groupedCampaigns);
     return groupedCampaigns ?? [];
 };
+
+const convertFormattedNumbers = body => {
```
Comment (beygorghor) [inline plugins/polio/js/src/hooks/useSaveCampaign.js:14]: Arg this is a bit ugly

**Review by beygorghor — CHANGES_REQUESTED** [review]
You can update `TextInput` by `NumberInput` in `plugins/polio/js/src/forms/RoundVaccineForm.tsx` i think

File: plugins/polio/js/src/hooks/useSaveCampaign.js (hunk 1 above)
Comment (quang-le) [reply plugins/polio/js/src/hooks/useSaveCampaign.js:14]: Yeah, but I don't see how else to do it without refactoring the whole thing, and I don't fancy doing that for putting commas in a number

**Comment by quang-le** [comment]
Closing for now as we will try a new approach using react-intl

**Pull request closed without merging**